x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_table.test.tsx

```diff
@@ -140,4 +172,75 @@ describe('EnginesTable', () => {
       expect(tableContent).not.toContain('Universal');
     });
   });
+
+  describe('actions', () => {
+    it('when AppLogic.values.myRole.canManageEngines is false actions are hidden', () => {
+      jest.clearAllMocks();
+      setMockValues({
+        // AppLogic
+        myRole: {
+          canManageEngines: false,
+        },
+      });
+      const wrapper = mountWithIntl(<EnginesTable {...props} />);
+      const tableRow = wrapper.find(EuiTableRow).first();
+
+      expect(tableRow.find(EuiIcon)).toHaveLength(0);
+    });
+
+    describe('when AppLogic.values.myRole.canManageEngines is true', () => {
+      let wrapper: ReactWrapper<any>;
+      let tableRow: ReactWrapper<any>;
+      let actions: ReactWrapper<any>;
+
+      beforeEach(() => {
+        jest.clearAllMocks();
+        setMockValues({
+          // AppLogic
+          myRole: {
+            canManageEngines: true,
+          },
+        });
+        wrapper = mountWithIntl(<EnginesTable {...props} />);
+        tableRow = wrapper.find(EuiTableRow).first();
+        actions = tableRow.find(EuiIcon);
+        EnginesLogic.mount();
+      });
+
+      it('renders a manage action', () => {
+        jest.spyOn(TelemetryLogic.actions, 'sendAppSearchTelemetry');
+        jest.spyOn(KibanaLogic.values, 'navigateToUrl');
+        actions.at(0).simulate('click');
+
+        expect(TelemetryLogic.actions.sendAppSearchTelemetry).toHaveBeenCalled();
+        expect(KibanaLogic.values.navigateToUrl).toHaveBeenCalledWith('/engines/test-engine');
+      });
+
+      describe('delete action', () => {
+        it('has an onClick action that calls window.confirm', () => {
+          jest.spyOn(global, 'confirm' as any).mockReturnValueOnce(true);
+          actions.at(1).simulate('click');
+          expect(global.confirm).toHaveBeenCalled();
+        });
+
+        it('clicking the action and confirming calls EnginesLogic.deleteEngine', () => {
+          jest.spyOn(global, 'confirm' as any).mockReturnValueOnce(true);
```

Comment:
If for some reason this *doesn't* get called in this test, does it bleed into the next test? I don't know the answer, I'm just thinking out loud

Reply:
Is this just a matter of calling `jest.clearAllMocks()`?

Reply:
Maybe! I'm not entirely sure off of the top of my head.

Reply:
Let's just move on, the tests pass, there are no issue.